### ISSUE TITLE
Scheduling early transition to Workbench

### ISSUE THREAD

**Issue by tobyhodges** [datacarpentry/openrefine-socialsci#140]
Short version: **are you are willing to volunteer your lesson for early transition to The Carpentries Workbench 
infrastructure?**

---

Longer version below:

As I hope you are already aware, @zkamvar and the rest of the Curriculum Team are preparing to roll out the new lesson infrastructure, [The Carpentries Workbench](https://carpentries.github.io/workbench/), across all of The Carpentries official lessons in early May 2023. This means that all Data Carpentry, Library Carpentry, and Software Carpentry lesson repositories will be modified to adopt the new infrastructure at the end of this month.

As you might imagine, coordinating a rollout like this involves a large amount of time and effort, and one thing that will really help us to keep to the schedule and avoid disruption for the community is to be able to **prepare some lessons for 
transition before the 1st May deadline**.

With this in mind, **I invite you to volunteer to schedule an early transition for theis lesson repository.** If you expect to have time to adopt the Workbench version of the lesson repository in the next couple of weeks, please reply to let me know. As lesson Maintainers, these are the things you will need to be prepared to do for the transition:
1. **Be prepared to quickly merge pull requests from Zhian**. These pull requests will help update the lesson and aid the transition process.
2. **Close any outstanding pull requests.** Your repository currently has very few or zero open PRs, but we cannot rule out the possibility that someone will open more in the time between now and when the transition takes place. [Open PRs will be invalidated when the transition takes place](https://carpentries.github.io/sandpaper-docs/pull-request.html#transition-from-carpentriesstyles).
3. **Preview the lesson (in the https://github.com/fishtree-attempt organisation) when it is available**. Zhian will open an issue in your repository, tagging you when the preview is available. You should reply on that issue if you notice any problems with the Workbench version of the lesson site and repository in the preview.
4. Just before the transition happens, the repository will be temporarily set as read-only and an issue will be opened (see an example from the Maintainer Onboarding curriculum: https://github.com/carpentries/maintainer-onboarding/issues/69) 
5. After the transition, **delete and recreate any forks and local clones** you have of the repository and then **confirm that you have done so** (you will be given specific instructions in an issue).

If you are willing and able to help us ensure a smooth rollout of the Workbench, please reply to let me know that you are happy to volunteer your lesson for early transition. If there is a specific date you would like the transition to take place (or that we should avoid), please also mention that.

**Comment by bencomp** [comment]
Hi @tobyhodges, I have responded to your email with suggested dates. When considering these dates, please note that I don't know when the lesson is being taught.

**Comment by zkamvar** [comment]
Thank you for participating in the early transition to The Workbench. I have been able to render a preview of your lesson at https://fishtree-attempt.github.io/openrefine-socialsci.

I am tracking issues for your lesson in https://github.com/carpentries/lesson-transition/issues/49. I will be updating the preview next week as I work on any remaining issues. 

Please inspect the preview and make note in https://github.com/carpentries/lesson-transition/issues/49 if you see anything that needs to be fixed (e.g extra templating, narrow tables, or malformed text).

**Comment by bencomp** [comment]
I added a few observations :green-sticky:

**Comment by tobyhodges** [comment]
Hi @bencomp we are getting ready for an early transition for this lesson, and we have prepared [a schedule for early lesson transitions](https://docs.google.com/spreadsheets/d/1Y_ZdAZu11Lc1bcInE9801ax6khvgFzjTh70P3v8crL8/edit?usp=sharing). 

Please check that document to see the date we have planned for transitioning this lesson, and adjust the sheet if a different date would be more convenient for you (no more than three lessons per day, please!). The README sheet gives more information. If we don't hear from you, we will go ahead with the transition on the date shown in that document.

**Comment by bencomp** [comment]
Looks good to me!

**Comment by zkamvar** [comment]
This lesson is ready for its transition tomorrow. There is one task to be done post-transition, which is detailed in https://github.com/carpentries/lesson-transition/issues/49#issuecomment-1523617764

**Comment by zkamvar** [comment]
The deed is done. The infrastructure takes a few minutes to bootstrap and cache the packages for the lesson build. Once the build is done, I will switch github pages to deploy from the `gh-pages` branch and you will have your workbench lesson. 

Thank you all for your enthusiasm and your patience!

**Comment by zkamvar** [comment]
The Workbench version is now live: https://datacarpentry.github.io/openrefine-socialsci/

In addition, here is [map of commits that were changed during the transition](https://github.com/carpentries/lesson-transition/blob/release_dc/openrefine-socialsci/release/datacarpentry/openrefine-socialsci-commit-map.hash)

**Comment by bencomp** [comment]
I am ready for write access :rocket: – and I think I may already have them 🟩

**Comment by zkamvar** [comment]
> – and I think I may already have them

😨 I had explicitly set the maintainer and curriculum advisors teams to use [the "Triage" role](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/repository-roles-for-an-organization):

> Triage: Recommended for contributors who need to proactively manage issues, discussions, and pull requests without write access

**Comment by zkamvar** [comment]
That being said, @bencomp, you now have write access!

**Comment by bencomp** [comment]
I was mistaken indeed. Thanks for your hard work, @zkamvar !

**Comment by bencomp** [comment]
@efran can I help you get ready for write access? I believe this issue can be closed as soon as the maintainers have write access. Thanks :)

**Comment by ErinBecker** [comment]
Hi @efran - You have not yet responded here to report that you completed the steps listed by @zkamvar above to regain your Maintainer access to this repository. The lesson infrastructure transition required a second Maintainer team to be temporarily created for all lessons, which can only be removed when all Maintainers have confirmed that they are ready to have their access restored. Maintenance of both teams is not sustainable for The Carpentries in the long run. At the beginning of **November 2023** (note the deadline has been extended) we will remove access for Maintainers who have not taken the necessary action to continue working on the repository.

The Curriculum Team is truly grateful for everything you have already done as a Maintainer and we do not want to lose you! But we also understand that circumstances change and you may no longer have capacity to continue in your role. If you wish to retain your access and keep supporting the community, please follow the steps above then confirm that you are ready by commenting I am ready for write access 🚀 on this thread before the end of your day on **Wednesday 1st November.**

**Comment by ErinBecker** [comment]
Hi @efran  - You no longer have Maintainer access to this repository. Please see the comment above for details. If you would like to restore your access, please re-clone the repository on your machine and then comment here with I am ready for write access 🚀 and the admin maintainer of this repository will restore your permissions. If you have any questions, please don’t hesitate to get in touch [curriculum@carpentries.org](mailto:curriculum@carpentries.org)

To the remaining Maintainers for this lesson - the Curriculum Team will be prioritizing onboarding of new Maintainers for lessons that have lost Maintainers in our next round of onboarding. Please don’t hesitate to reach out if your lesson needs additional support in the meantime - [curriculum@carpentries.org](mailto:curriculum@carpentries.org).

I will close this issue now but will still be notified of any replies on the thread in future.
